air/src/proof/ood_frame.rs

```diff
@@ -38,29 +46,48 @@ impl OodFrame {
     // --------------------------------------------------------------------------------------------
 
     /// Updates the trace state portion of this out-of-domain frame. This also returns a compacted
-    /// version of the out-of-domain frame with the rows interleaved. This is done so that reseeding
-    /// of the random coin needs to be done only once as opposed to once per each row.
+    /// version of the out-of-domain frame (including the Lagrange kernel frame, if any) with the
+    /// rows interleaved. This is done so that reseeding of the random coin needs to be done only
+    /// once as opposed to once per each row.
     ///
     /// # Panics
     /// Panics if evaluation frame has already been set.
-    pub fn set_trace_states<E: FieldElement>(&mut self, trace_states: &[Vec<E>]) -> Vec<E> {
+    pub fn set_trace_states<E: FieldElement>(
+        &mut self,
+        trace_states: &OodFrameTraceStates<E>,
+    ) -> Vec<E> {
```

Comment:
I actually like that an Ood frame contains both the "regular" current/next frames, as well as the Lagrange kernel frame. I think of it as one unit; all the data necessary to check the out-of-domain point in one place. I'm curious why you were thinking about separating them though.

> (with this, we may be able to avoid the need for `OodFrameTraceStates` struct)

How do you see we get rid of it? We'll still need to store the current & next frames.

But I also have as mental note to reduce the number of "ood frame" structs, but I was leaving it for another PR :).